docs/programming_guide/execution_api_type/client_api.rst

```diff
@@ -167,20 +167,26 @@ Client API communication patterns
 
 We offer various implementations of Client APIs tailored to different scenarios, each linked with distinct communication patterns.
 
-Broadly, we present in-process and sub-process executors. The in-process executor, slated for release in NVFlare 2.5.0,
-entails both training scripts and client executor operating within the same process. The training scripts will be launched once
-at the event of START_RUN. The training scripts keep on running till the END_RUN event. Communication between them occurs
-through an in-memory databus.
+In-process Client API
+---------------------
 
-On the other hand, the LauncherExecutor employs a sub-process to execute training scripts, leading to the client executor
-and training scripts residing in separate processes. The "launch_once" option is provided to the SubprocessLauncher to control
+The in-process executor entails both the training script and client executor operating within the same process.
+The training script will be launched once at the event of START_RUN and will keep on running till the END_RUN event.
+Communication between them occurs through an efficient in-memory databus.
+
+When the training process involves either a single GPU or no GPUs, and the training script doesn't integrate third-party
+training systems, the in-process executor is preferable (when available).
+
+Sub-process Client API
+----------------------
+
+On the other hand, the LauncherExecutor employs a sub-process to execute training script, leading to the client executor
+and training script residing in separate processes. The "launch_once" option is provided to the SubprocessLauncher to control
```

Comment:
A note here the `LauncherExecutor` itself is not using subprocess, it is using the `Launcher` abstraction.

The `SubprocessLauncher` is using subprocess to launch external process.

Some of these abstract/classes we might consolidate with Yan's new PR.